nautobot_ssot/migrations/0007_replace_dashed_custom_fields.py

```diff
@@ -0,0 +1,53 @@
+from django.db import migrations
+from nautobot.extras.utils import FeatureQuery
+
+CF_KEY_CHANGE_MAP = {
+    "servicenow_last_synchronized": "ssot-synced-to-servicenow",
+    "last_synced_from_sor": "ssot-synced-from-ipfabric",
+    "ipfabric_site_id": "ipfabric-site-id",
+    "ssot_synced_to_infoblox": "ssot-synced-to-infoblox",
+}
+
+
+def replace_dashed_custom_fields(apps, schema_editor):
+    CustomField = apps.get_model("extras", "customfield")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    for new_key, old_key in CF_KEY_CHANGE_MAP.items():
+        if not CustomField.objects.filter(key=new_key).exists():
+            for custom_field in CustomField.objects.filter(key=old_key):
+                original_key = custom_field.key
+                print(
+                    f'   CustomField instance "{custom_field.label}" key attribute "{original_key}" is being changed to "{new_key}".'
+                )
+                custom_field.key = new_key
+                custom_field.save()
+
+    for ct in ContentType.objects.filter(FeatureQuery("custom_fields").get_query()):
+        model = apps.get_model(ct.app_label, ct.model)
+        cf_list = []
+        for instance in model.objects.all():
```

Comment:
Concerned about the performance overhead here of iterating through literally every instance of every model that supports any custom fields. Can we at least narrow this down to just the set of models that the old CFs from this app would exist for - looks like that'd be `[Device, DeviceType, Interface, Manufacturer, Location, VLAN, Role, IPAddress]`?

Reply:
Done.